functions/Utils.js

```diff
@@ -1,21 +1,127 @@
+            // Set path to primary contact email
+            path = ('accounts/' + snapshot.val() + '/email');
+            // Send an email with updated path
+            admin.database().ref(path).once('value').then(snapshot => {
+                return emailNotification(snapshot.val(), templateId);
+            })
+                .catch(err => {
+                    console.log(err);
+                });  
+            return path;
+        })
+            .catch(err => {
+                console.log(err);
+            });  
+        return promise;
+    }
+
+    admin.database().ref(path).once('value').then(snapshot => {
+        return emailNotification(snapshot.val(), templateId);
+    })
+        .catch(err => {
+            console.log(err);
+        });  
     return promise;
 };
 
 // Create a notification obj for the given pickup request and notification
 // type and pushes it to the account
-var notifyRequestUpdate = function (label, acctRef, requestPath, notifType) {
+var notifyRequestUpdate = function (label, acctRef, requestPath, notifType, templateId) {
     let notification = {
         type: notifType,
         content: requestPath
     };
-    return pushNotification(label, acctRef, notification);
+    return pushNotification(label, acctRef, notification, templateId);
 };
 
+// Takes in the email contact to be emailed and also which sendgrid email template to use
+// Returns an email message object based on what emailContent is passed.
+var emailNotification = function (email, templateId) {
+    // templateID = String(templateId);
+    // email to be replaced with email parameter
+    let gmail = functions.config().email_login.email;
+    let password = functions.config().email_login.password;
+    var transporter = nodemailer.createTransport({
+        service: 'gmail',
+        host: 'smtp.gmail.com',
+        auth: {
+            user: gmail,
+            pass: password,
+        }
+    });
+        
+    var mailOptions = selectTemplates(templateId, email);       
+    transporter.sendMail(mailOptions, function(error, info){
+        if (error) {
+            console.log(error);
+        } else {
+            console.log('Email sent: ' + info.response);
+        }
+    });
+    return;
+};
+
+var selectTemplates = function (templateId, email) {
+    console.log(email);
+    if(templateId === 'email_all') {
+        email_content = {
+            from: 'noreplymealmatchup.com',
+            to: email,
+            subject: 'Meal Matchup Delivery Confirmation',
+            html:'<html> <head> <title></title> </head> <body><p>Hello,</p><div> </div><span>The delivery that you are a part of has been confirmed by all 3 parties. To see the updated status of the delivery, please login here: </span><a href="https://www.mealmatchup.org/">https://www.mealmatchup.org/</a><span>.</span><p>Thank you!</p><p>Meal Matchup</p></body></html>'
+        };
+    } else if(templateId === 'no_available_RAs') {
```

Comment:
I think a better idea for these templateId's would be to use an enum rather than just a plain string. We can talk about it in our next meeting

Reply:
Just updated using Enums. Let me know if this is the approach or if you want me to change it slightly. (Originally I was just trying to get the whole dictionary and then push the email onto it but that wasn't working for some reason and firebase wasn't updating so it was hard to test)